core/src/main/java/org/apache/iceberg/TableProperties.java

```diff
@@ -140,4 +140,10 @@ private TableProperties() {
 
   public static final String GC_ENABLED = "gc.enabled";
   public static final boolean GC_ENABLED_DEFAULT = true;
+
+  public static final String WRITE_ISOLATION_LEVEL = "write.isolation-level";
```

Comment:
Well, I wanted it to be `write.delete.isolation-level` but we have no way to find out which operation is being performed in the merge builder.

Reply:
It might be better to use `write.row-level.isolation-level` so it matches the mode property. This doesn't apply to `INSERT OVERWRITE` so it would be nice to be clear about it.

Not sure if we can make that property name better. Seems odd to have `row-level` and `isolation-level` next to one another since "row-level" is specifying the granularity of the changes and "isolation-level" is actually choosing an option.

Reply:
Fixed.